### ISSUE TITLE
Increasing batch size is not reducing no. of iterations in an epoch

### ISSUE THREAD

**Issue by debnay** [junyanz/pytorch-CycleGAN-and-pix2pix#1025]
Hi,
I am new to CycleGAN and am just getting started with the code. I am trying to run the dataset maps through it, just as outlined in the 'Getting Started' section. I have kept all the parameters same, except for --display_id -1 and --batch_size 4. Should this not reduce the no. of iterations per epoch? Whether I use --norm batch (batch normalization) or just let it be (instance normalization), the no. of iterations / epoch keeps going to 1000 (approx.) [No. of training images in maps = 1096]. Am I missing something?

**Comment by junyanz** [comment]
The iteration counter takes the `bach_size` as consideration. See this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/train.py#L76). Feel free to change it.

**Comment by debnay** [comment]
I see. But I do not see significant improvement in 'Time Taken' for each epoch by increasing batch size.
--batch_size 1: Avg. 'Time Taken'/epoch - 1382s
--batch_size 2: Avg. 'Time Taken'/epoch - 1443s
--batch_size 2 --norm batch: Avg. 'Time Taken'/epoch - 1278s
--batch_size 4: Avg. 'Time Taken'/epoch - 1393s
--batch_size 4 --norm batch: Avg. 'Time Taken'/epoch - 1215s
Is this expected?

**Comment by junyanz** [comment]
You will get a small improvement but not huge, as you are still using a single GPU. You may also want to check the data loading time and see if it is a bottleneck. You can increase the `--num_threads` to reduce data loading time.

**Comment by debnay** [comment]
I did try experimenting with --num_threads. But I did not see any significant boost. So I guess the default --num_threads 4 is working fine. Thanks a lot for your insights.
Also, eventually I would be running my experiments in multi-gpu mode. 
1. In your experience does instance normalization work best in such a scenario? 
2. No matter what --batch_size is given, the first GPU will always be loaded to the maximum extent possible and then the next GPU will take over? Or, all the images in --batch_size will be evenly distributed among the GPUs?

**Comment by junyanz** [comment]
1. By performance, I assume that you mean image quality. batch_size=1 works quite well, and a big batch size works quite well (like 16 or 32). A smaller batch size (2 or 4) might not be helpful. 
2. There is no clear winner. But our batch normalization is not synchronized across GPUs. If you want to use batch norm with multiple GPUs, you need to adopt this [repo](https://github.com/vacancy/Synchronized-BatchNorm-PyTorch). A recent work StyleGANv2 also shows that normalization might hurt the performance. 
3. They are equally distributed.
